lib/adbc_connection.ex

```diff
@@ -395,7 +395,16 @@ defmodule Adbc.Connection do
     end
   end
 
-  defp create_statement(conn, query) do
+  defp create_statement(conn, query) when is_pid(conn) do
+    %{conn: conn} = :sys.get_state(conn)
```

Comment:
Trying to learn a bit about the internals — @cocoa-xu what does this new clause do and is the `when is_reference(conn)` clause still needed?

Reply:
Hi @dlindenkreuz, this is because 

```elixir
conn = start_supervised!({Connection, database: db})
```

returns a GenServer process, which we can test it with `is_pid/1`. 

And the real underlying `Adbc.Connection.t()` is allocated in [`Adbc.Connection.start_link/1`](https://github.com/elixir-explorer/adbc/blob/3e6ebdcbd79d1a149579cbeed64ad34e2334ef12/lib/adbc_connection.ex#L41) and stored as an internal state of the GenServer process.

Therefore, we can use `:sys.get_state/1` to retrieve the internal state of a GenServer process so that we can get the real `Adbc.Connection.t()`:)